expertsystem/state/propagation.py

```diff
@@ -199,6 +219,209 @@ def find_solutions(self):
         return full_particle_graphs
 
 
+def _is_optional(class_field):
+    if "__args__" in class_field.__dict__:
+        if type(None) in class_field.__args__:
+            return True
+    return False
+
+
+_qn_mapping = {
```

Comment:
hmm since only us two are going to be confronted with this, i did not think that is necessary. I will be removed with the next propagation code refactoring

Reply:
Ok